indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/autoscaler/CostBasedAutoScalerConfig.java

```diff
@@ -0,0 +1,410 @@
+  private final int taskCountMax;
+  private final int taskCountMin;
+  private Integer taskCountStart;
+  private final long minTriggerScaleActionFrequencyMillis;
+  private final Double stopTaskCountRatio;
+
+  private final long metricsCollectionIntervalMillis;
+  private final long metricsCollectionRangeMillis;
+  private final long scaleActionStartDelayMillis;
+  private final long scaleActionPeriodMillis;
+
+  private final double lagWeight;
+  private final double idleWeight;
+
+  @JsonCreator
+  public CostBasedAutoScalerConfig(
+      @JsonProperty("taskCountMax") Integer taskCountMax,
+      @JsonProperty("taskCountMin") Integer taskCountMin,
+      @Nullable @JsonProperty("enableTaskAutoScaler") Boolean enableTaskAutoScaler,
+      @Nullable @JsonProperty("taskCountStart") Integer taskCountStart,
+      @Nullable @JsonProperty("minTriggerScaleActionFrequencyMillis") Long minTriggerScaleActionFrequencyMillis,
+      @Nullable @JsonProperty("stopTaskCountRatio") Double stopTaskCountRatio,
+      @Nullable @JsonProperty("metricsCollectionIntervalMillis") Long metricsCollectionIntervalMillis,
+      @Nullable @JsonProperty("metricsCollectionRangeMillis") Long metricsCollectionRangeMillis,
+      @Nullable @JsonProperty("scaleActionStartDelayMillis") Long scaleActionStartDelayMillis,
+      @Nullable @JsonProperty("scaleActionPeriodMillis") Long scaleActionPeriodMillis,
+      @Nullable @JsonProperty("lagWeight") Double lagWeight,
+      @Nullable @JsonProperty("idleWeight") Double idleWeight
+  )
+  {
+    this.enableTaskAutoScaler = enableTaskAutoScaler != null ? enableTaskAutoScaler : false;
+
+    // Timing configuration with defaults
+    this.metricsCollectionIntervalMillis = metricsCollectionIntervalMillis != null
+                                           ? metricsCollectionIntervalMillis
+                                           : DEFAULT_METRICS_COLLECTION_INTERVAL_MILLIS;
+    this.metricsCollectionRangeMillis = metricsCollectionRangeMillis != null
+                                        ? metricsCollectionRangeMillis
+                                        : DEFAULT_METRICS_COLLECTION_RANGE_MILLIS;
+    this.scaleActionStartDelayMillis = scaleActionStartDelayMillis != null
+                                       ? scaleActionStartDelayMillis
+                                       : DEFAULT_SCALE_ACTION_START_DELAY_MILLIS;
+    this.scaleActionPeriodMillis = scaleActionPeriodMillis != null
+                                   ? scaleActionPeriodMillis
+                                   : DEFAULT_SCALE_ACTION_PERIOD_MILLIS;
+    this.minTriggerScaleActionFrequencyMillis = minTriggerScaleActionFrequencyMillis != null
+                                                ? minTriggerScaleActionFrequencyMillis
+                                                : DEFAULT_MIN_TRIGGER_SCALE_ACTION_FREQUENCY_MILLIS;
+
+    // Cost function weights with defaults
+    this.lagWeight = lagWeight != null ? lagWeight : DEFAULT_LAG_WEIGHT;
```

Comment:
Use utility method for brevity
```suggestion
    this.lagWeight = Configs.valueOrDefault(lagWeight, DEFAULT_LAG_WEIGHT);
```